### PR TITLE
Gulp Tests now report summaries even when some tests fail.

### PR DESCRIPTION
**Pull request by alexbragdon**
Address issue #5539.

Created safe versions of tests for Common, API, Karma, and E2E in gulp-tests.js which no longer callback their errors. 

Removed the pushing to testResults by the non-safe versions of these tests since they will now only be run independently. 

Lastly, after the summary is displayed an error will be thrown if any of the tests did fail, thus preserving the operation of Travis CI.

### CODE REVIEW

File: tasks/gulp-tests.js [hunk 1]
```diff
@@ -164,5 +219,8 @@ gulp.task('test', [
     `\x1b[36mPending: ${totals[2]}\t`
   );
 
-  console.log('\n\x1b[36mThanks for helping keep Habitica clean!\x1b[0m');
+	if (totals[1] > 0) throw "ERROR: There are failing tests!"
+	else {
+		console.log('\n\x1b[36mThanks for helping keep Habitica clean!\x1b[0m');
+	}
```
Comment (gisikw) [inline tasks/gulp-tests.js]: Really minor thing - mind using soft tabs here (2 spaces)?

**Comment by gisikw** [comment]
This looks great otherwise! Will be really helpful for folks to be able to review the summary results (since running the suites asynchronously can make the output a bit hard to parse)

**Comment by alexbragdon** [comment]
My bad, wasn't paying attention to formatting.

**Comment by gisikw** [comment]
No worries! Looks good - I'll merge this in pending CI :+1:

//cc @SabreCat - Proper test summaries even when a particular suite fails. Should mean you can use _some_ of the suites in a Windows env now :)

**Comment by gisikw** [comment]
w00t, thanks for this! This'll be really helpful, especially for the folks working in the Vagrant environment :)

//cc @crookedneighbor @Alys for contributor stuffs

**Comment by crookedneighbor** [comment]
@alexbragdon This will count towards a contributor tier! What's your User ID? You can find it in [Settings > API](https://habitrpg.com/#/options/settings/api). Be sure to only copy the User ID and _not_ your API Token.

**Comment by alexbragdon** [comment]
Cool, sure thing;

User ID: a3835f07-f60c-406d-9321-888a5d57e8ec

**Comment by crookedneighbor** [comment]
Congratulations on your first tier!
